controllers/tasksRequests.js

```diff
@@ -43,6 +46,73 @@ const fetchTaskRequestById = async (req, res) => {
   }
 };
 
+const addTaskRequests = async (req, res) => {
+  try {
+    const taskRequestData = req.body;
+    const usernamePromise = usersUtils.getUsername(taskRequestData.userId);
+    if (req.userData.id !== taskRequestData.userId && !req.userData.roles?.super_user) {
+      return res.boom.forbidden("Not authorized to create the request");
+    }
+    if (taskRequestData.proposedDeadline < taskRequestData.proposedStartDate) {
+      return res.boom.badRequest("Task deadline cannot be before the start date");
+    }
+    switch (req.body.requestType) {
+      case TASK_REQUEST_TYPE.ASSIGNMENT: {
+        const taskDataPromise = tasksModel.fetchTask(taskRequestData.taskId);
+
+        const [{ taskData }, username] = await Promise.all([taskDataPromise, usernamePromise]);
+        taskRequestData.taskTitle = taskData?.title;
+        if (!username) {
+          return res.boom.badRequest("User not found");
+        }
+        if (!taskData) {
+          return res.boom.badRequest("Task does not exist");
+        }
+        break;
+      }
+      case TASK_REQUEST_TYPE.CREATION: {
+        let issuePromise;
+        try {
+          const url = new URL(taskRequestData.externalIssueUrl);
+          const issueUrlPaths = url.pathname.split("/");
+          const repositoryName = issueUrlPaths[3];
+          const issueNumber = issueUrlPaths[5];
+          issuePromise = githubService.fetchIssuesById(repositoryName, issueNumber);
```

Comment:
- If there's any Error in line [76 - 79](https://github.com/Real-Dev-Squad/website-backend/pull/1610/files#diff-44a86e7455eff885c3dceb5ffac512f66e4d862e64635704dd77f4d54e8c1548R76-R79), it would return `External issue URL is not valid`

- if there's any Error in `githubService.fetchIssuesById` function it will return `Issue does not exist` in line [90](https://github.com/Real-Dev-Squad/website-backend/pull/1610/files#diff-44a86e7455eff885c3dceb5ffac512f66e4d862e64635704dd77f4d54e8c1548R90)

Am I missing anything?

Reply:
This will happen if `fetchIssuesById` does not throw an error and returns null. 
But as its logic is entirely based on making a fetch call to GitHub API, it would throw an error if it fails to fetch due to any reason. And then it will get caught and the server will return `Error while creating task request`.
Maybe adding regex while validating the `body.externalIssueUrl` will make sure the initials of the provided URL are as expected.

Please correct me if I am wrong.

Reply:
Added regex validation.


Not sure about what you are trying to explain. Can you please explain it with a test case example?